File: tracknstock-frontend/src/Components/Product.ts
export interface IProduct {
    id: number;
    name: string;
    boughtPrice: number;
    unitPrice: number;
    quantity: number;
    totalPrice: number;
}

export const dummyProductList: IProduct[] = [
    {
    id: 1,
    name: "Product A",
    boughtPrice: 50,
    unitPrice: 70,
    quantity: 10,
    totalPrice: 700
    },
]

export enum PageEnum {
    list,
    add
}

File: TrackNStock.Api/Models/DomainModel/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TrackNStock.Api.Models.DomainModel
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int BoughtPrice { get; set; }
        public decimal UnitPrice { get; set; }
        public int Quantity { get; set; }
        public string? Description { get; set; }
        public decimal TotalPrice => UnitPrice * Quantity;
        public bool Sold { get; set; } = false;
    }
}